src/main/java/io/confluent/connect/hdfs/hive/HiveMetaStore.java

```diff
@@ -71,19 +71,10 @@ public HiveMetaStore(Configuration conf, HdfsSinkConnectorConfig connectorConfig
     R call() throws TException;
   }
 
-  private <R> R doWithRetry(ClientAction<R> action) throws TException {
-    try {
-      return action.call();
-    } catch (TException e) {
-      try {
-          client.reconnect();
-      } catch (MetaException swallowedException) {
-        // reconnect failed, throw the original exception
-        log.warn("Cannot connect to Hive meta store: {}", swallowedException.getMessage());
-        throw e;
-      }
-      return action.call();
-    }
+  private <R> R doAction(ClientAction<R> action) throws TException {
+    // No need to implement retries. We use RetryingHiveMetaStoreClient
+    // which already implements the retrying and reconnect logic
```

Comment:
We hit the `AlreadExistsException` (which is a type of `TException`: https://hive.apache.org/javadocs/r0.10.0/api/org/apache/hadoop/hive/metastore/api/AlreadyExistsException.html) very often. This causes the `reconnect` to be called multiple times. Snippet from the logs of our forked code:

```
[2016-05-11 03:31:55,353] [pool-13-thread-1] **WARN [DEBBUG] Action failed. Reconnecting. (io.confluent.connect.hdfs.hive.HiveMetaStore:80)**
AlreadyExistsException(message:Partition already exists:Partition(values:[2016-05-10, 1462849200], dbName:xxxxx, tableName:client_timi
```

We package against the JARs specified in `kafka-connect-hdfs` pom.xml: So I'm guessing the Hive version is 1.2.1? Our deployed Hive version is also 1.2.1.